dnsseed.go

```
@@ -0,0 +1,211 @@
+				amgr.AddAddresses(addrs)
+			})
+			peers = amgr.Addresses()
+		}
+		if len(peers) == 0 {
+			log.Printf("No stale addresses -- sleeping for 10 minutes")
+			for i := 0; i < 600; i++ {
+				time.Sleep(time.Second)
+				if atomic.LoadInt32(&systemShutdown) != 0 {
+					log.Printf("Creep thread shutdown")
+					return
+				}
+			}
+			continue
+		}
+
+		for _, addr := range peers {
+			if atomic.LoadInt32(&systemShutdown) != 0 {
+				log.Printf("Waiting creep threads to terminate")
+				wgCreep.Wait()
+				log.Printf("Creep thread shutdown")
+				return
+			}
+			wgCreep.Add(1)
+			go func(addr *wire.NetAddress) {
+				defer wgCreep.Done()
+
+				host := net.JoinHostPort(addr.IP.String(), strconv.Itoa(int(addr.Port)))
+				p, err := peer.NewOutboundPeer(&config, host)
+				if err != nil {
+					log.Printf("NewOutboundPeer on %v: %v",
+						host, err)
+					return
+				}
+				amgr.Attempt(addr.IP)
+				conn, err := net.DialTimeout("tcp", p.Addr(), nodeTimeout)
+				if err != nil {
+					log.Printf("%v", err)
+					return
+				}
+				p.AssociateConnection(conn)
+
+				// Wait version messsage or timeout in case of failure.
+				select {
+				case <-onVersion:
+				case <-time.After(nodeTimeout):
+					log.Printf("version timeout on peer %v",
+						p.Addr())
+					p.Disconnect()
+					return
+				}
+
+				select {
+				case <-onAddr:
+				case <-time.After(nodeTimeout):
+					log.Printf("getaddr timeout on peer %v",
+						p.Addr())
+					p.Disconnect()
+					return
+				}
+				p.Disconnect()
+			}(addr)
+		}
+		wgCreep.Wait()
+	}
+}
+
+func main() {
+	cfg, err := loadConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "loadConfig: %v\n", err)
+		os.Exit(1)
+	}
+	amgr, err = NewManager(defaultHomeDir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "NewManager: %v\n", err)
+		os.Exit(1)
+	}
+
+	peersDefaultPort, _ = strconv.Atoi(activeNetParams.DefaultPort)
```

why do you ignore the error?

---

DefaultPort must be checked by loadConfig, called above.